File: tests/fixtures/realistic/js_multihop_sanitized/service.js
// Middle helper (fileB) for the js_multihop_sanitized fixture.
//
// Unlike the positive fixture, handle() runs the tainted value through
// mysql.escape() — a configured sanitizer for js/taint-sql-injection — before
// forwarding it to store.runQuery(). The sanitizer collapses the value to
// "clean", so the composed summary must NOT record a params_to_sink flow, and
// the chain breaks: no taint finding on a directory scan.

const store = require("./store");
const mysql = { escape(v) { return v; } };

function handle(term) {
    const safe = mysql.escape(term);
    return store.runQuery(safe);
}

module.exports = { handle };

File: tests/fixtures/realistic/python_multihop_sanitized/service.py
# Middle helper (fileB) for the python_multihop_sanitized fixture.
#
# Unlike the positive fixture, handle() runs the tainted value through
# escape_string() — a configured sanitizer for py/taint-sql-injection — before
# forwarding it to db.run_query(). The sanitizer collapses the value to "clean",
# so the composed summary must NOT record a params_to_sink flow, and the chain
# breaks: no taint finding on a directory scan.

from . import db


def handle(term):
    safe = escape_string(term)
    return db.run_query(safe)
